src/main/java/com/gmail/val59000mc/scenarios/scenariolisteners/MonstersIncListener.java

```diff
@@ -0,0 +1,83 @@
+package com.gmail.val59000mc.scenarios.scenariolisteners;
+
+import com.gmail.val59000mc.UhcCore;
+import com.gmail.val59000mc.scenarios.ScenarioListener;
+
+import java.util.List;
+import java.util.HashMap;
+
+import org.bukkit.Location;
+import org.bukkit.entity.Player;
+import org.bukkit.event.player.PlayerInteractEvent;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockFace;
+import org.bukkit.block.data.Bisected;
+import org.bukkit.event.block.Action;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.block.BlockPlaceEvent;
+import org.bukkit.event.block.BlockBreakEvent;
+import org.bukkit.metadata.FixedMetadataValue;
+import org.bukkit.metadata.MetadataValue;
+
+public class MonstersIncListener extends ScenarioListener {
+
+    UhcCore plugin = UhcCore.pl;
+    private int DoorsPlaced;
+    private HashMap<Integer, Location> DoorLocs = new HashMap<>();
+
+    @EventHandler (ignoreCancelled = true)
+    public void onBlockPlace(BlockPlaceEvent e) {
+
+        Block block = e.getBlock();
+        Location loc = e.getBlock().getLocation();
+
+        if(!block.getType().toString().contains("TRAP") && block.getType().toString().contains("DOOR")) {
```

Comment:
The benefit I see in comparing strings is that future door types are automatically supported. I also intentionally made trapdoors usable with this if wanted as stated in Consideration 1. If it's an issue, I can change this.

Reply:
Hmmm... I understand what you mean but at the same time new door types or items containing "door" in their name could break the scenario. Maybe move the if statement to a `isDoor(Block)` method so if needed it could be changed easily in the future.

Reply:
String check is now a boolean statement as of 3b19eee